asana/asana.go

```diff
@@ -189,8 +172,35 @@ type (
 
 	// Errors always has at least 1 element when returned.
 	Errors []Error
+
+	// HTTP request error
+	requestError struct {
+		Body string
+		Code int
+	}
 )
 
+func (re requestError) Error() string {
+	txt := "request error"
+	switch re.Code {
+	case http.StatusBadRequest:
+		txt = "bad request"
+	case http.StatusUnauthorized:
+		txt = "unauthorized"
+	case http.StatusPaymentRequired:
+		txt = "payment required"
+	case http.StatusForbidden:
+		txt = "forbidden"
+	case http.StatusNotFound:
+		txt = "not found"
+	case http.StatusTooManyRequests:
+		txt = "too many requests"
+	case http.StatusInternalServerError:
+		txt = "internal server error"
+	}
+	return fmt.Sprintf("asana: Error: %s, got HTTP response code %d with body: %v", txt, re.Code, re.Body)
```

Comment:
I'd probably leave the body out of the error message since it can be large. Require callers to access the body if they want details. The googleapi error uses the body if the message is not set.

https://github.com/googleapis/google-api-go-client/blob/master/googleapi/googleapi.go#L89

Reply:
Removed it.